python/tank/pipelineconfig_factory.py

```diff
@@ -279,29 +278,38 @@ def _get_configuration_context():
         # Once found, we can double check that the current Entity is actually
         # associated with the project that the PC is associated with.
         val = pipelineconfig_utils.get_config_install_location(curr_pc_path)
-        
+
     return val
 
 
+def _cleanup_separators(path):
+    """
+    Flips all slashes to the operating system's standard slash separator and strips
```

Comment:
I am not sure I see the benefit of having a method that contains one line of code and is only used in one place. This is my personal opinion, but wondering it's just overkill? Maybe instead:

```python
def _get_pipeline_configuration_paths(sg_pipeline_configs):
    """
    Given a list of Shotgun Pipeline configuration entity data, return a list
    of pipeline configuration paths for the current platform. Also returns
    the local os path to any primary pipeline configurations if such are found.
    :param sg_pipeline_configs: SG pipeline configuration data. List of dicts.
    :returns: (all_paths, primary_paths) - tuple with two lists of path strings
    """
    # get list of local path to pipeline configurations that we have
    platform_lookup = {"linux2": "linux_path", "win32": "windows_path", "darwin": "mac_path" }
    local_pc_paths = []
    primary_pc_paths = []
    for pc in sg_pipeline_configs:
        curr_os_path = pc.get(platform_lookup[sys.platform])
        # flip all slashes to the operating system's standard slash separator
        curr_os_path = curr_os_path.replace("\\", os.path.sep).replace("/", os.path.sep)
        # strip any slashes or whitespace at the end
        curr_os_path = curr_os_path.rstrip("\\ /")
        local_pc_paths.append(curr_os_path)
        if pc.get("code") == constants.PRIMARY_PIPELINE_CONFIG_NAME:
            primary_pc_paths.append(curr_os_path)

    return (local_pc_paths, primary_pc_paths)
```